### PR TITLE
Vision

### PR DESCRIPTION
**Pull request by Rylan-Meilutis**
Added pathing and all the chassis stuff

### CODE REVIEW

**Review by Rylan-Meilutis — COMMENTED** [review]
Looks good to me

File: build.gradle
Comment (mmarchetti) [inline build.gradle:3]: It doesn't seem like we should downgrade this

File: paths/pythonParser.py
Comment (mmarchetti) [inline paths/pythonParser.py]: `fileName[:-4]` also works

File: paths/pythonParser.py
Comment (mmarchetti) [inline paths/pythonParser.py]: I sense some easy speedup opportunities that we should take advantage of before merging

File: paths/pythonParser.py
Comment (mmarchetti) [inline paths/pythonParser.py]: Similarly, `fileName[6:-4]` is equivalent

File: paths/pythonParser.py
Comment (mmarchetti) [inline paths/pythonParser.py]: This would benefit from being split out into a separate function or method

File: paths/pythonParser.py
Comment (mmarchetti) [inline paths/pythonParser.py]: raw strings will make your paths much nicer: 
`r"\src\main\java\frc\robot\routes\"`

File: src/main/java/frc/robot/DTXboxController.java
Comment (mmarchetti) [inline src/main/java/frc/robot/DTXboxController.java:80]: has this file been through the java formatter? this seems like a lot of formatting change

File: src/main/java/frc/robot/Wiring.java
Comment (mmarchetti) [inline src/main/java/frc/robot/Wiring.java:14]: Why did this change? Did we reprogram the motor controllers with new IDs?

File: src/main/java/frc/robot/components/DevilDrive.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/DevilDrive.java:138]: I'm a little surprised to see changes in this file, since it started out identical to the WPILib MecanumDrive and then we changed the motors to velocity mode. Should these new methods be in the Chassis instead?

File: src/main/java/frc/robot/components/IMU.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/IMU.java:74]: It might be good to put TotalYaw and YawRateDPS on the dashboard while we are debugging

File: src/main/java/frc/robot/components/MachineLearning.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/MachineLearning.java:69]: If we're writing to a file, should we write the line immediately? What's the value in buffering to the ArrayLists?

File: src/main/java/frc/robot/components/Vision.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/Vision.java:68]: Is this used anywhere?

File: src/main/java/frc/robot/components/VisionData.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/VisionData.java:19]: I don't love the extra lines before the `else` clauses, but I'll leave it to the team to decide

File: src/main/java/frc/robot/subsystems/Chassis.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/Chassis.java:29]: naming could be improved

File: src/main/java/frc/robot/subsystems/Chassis.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/Chassis.java]: `r` is passed in, then set from the sparkmax instead, then discarded? Can you help me understand what you're trying to do?

File: src/main/java/frc/robot/subsystems/Chassis.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/Chassis.java:88]: Why the 0.995s?

File: src/main/java/frc/robot/subsystems/Chassis.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/Chassis.java]: why the 0.5s?

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: I think we decided that hoop will also use only rotation to aim (not drive forward or backward). Range to target will determine the shooter wheel speed.

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: seems like `hoop_sidespeed` should also always be 0

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: this code could use more structure (separate methods with clearly defined functionality)

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: once hoop forward driving goes away, this method could be simplfied and renamed to something like `calculateHoopRotation`. Then it could return the calculated value instead of setting fields on the object, which is usually easier to think about and debug.

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: Similarly, this method can return the calculated value. We don't need to store it.

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (mmarchetti) [inline src/main/java/frc/robot/subsystems/VisionControl.java]: Seems like this method should be eliminated in favor of printing the relevant value (ball or hoop rotation) when it is calculated.

**Review by mmarchetti — CHANGES_REQUESTED** [review]
I had a lot of feedback on this PR because there is a lot of new code here. Let's discuss at the meeting.

File: build.gradle
Comment (Rylan-Meilutis) [reply build.gradle:3]: we will upgrade that after we flash the roborios

File: paths/pythonParser.py
Comment (Rylan-Meilutis) [reply paths/pythonParser.py]: done

File: paths/pythonParser.py
Comment (Rylan-Meilutis) [reply paths/pythonParser.py]: it adds to the experience

File: paths/pythonParser.py
Comment (Rylan-Meilutis) [reply paths/pythonParser.py]: done

File: paths/pythonParser.py
Comment (Rylan-Meilutis) [reply paths/pythonParser.py]: done

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (Rylan-Meilutis) [reply src/main/java/frc/robot/subsystems/VisionControl.java]: done

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (Rylan-Meilutis) [reply src/main/java/frc/robot/subsystems/VisionControl.java]: done

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (Rylan-Meilutis) [reply src/main/java/frc/robot/subsystems/VisionControl.java]: done

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (Rylan-Meilutis) [reply src/main/java/frc/robot/subsystems/VisionControl.java]: done

File: src/main/java/frc/robot/subsystems/VisionControl.java
Comment (Rylan-Meilutis) [reply src/main/java/frc/robot/subsystems/VisionControl.java]: done

File: paths/pythonParser.json
Comment (mmarchetti) [inline paths/pythonParser.json:10]: will this work as a local path (`./pythonParser.py`)?

File: paths/pythonParser.json
Comment (mmarchetti) [inline paths/pythonParser.json:22]: if there's an icon, it should live inside this repository

File: src/main/java/frc/robot/components/DevilDrive.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/DevilDrive.java:68]: According to [the SparkMax Java docs](https://codedocs.revrobotics.com/java/com/revrobotics/sparkmaxpidcontroller#setReference(double,com.revrobotics.CANSparkMax.ControlType)), the velocity reference value should be in RPM.

File: src/main/java/frc/robot/components/MachineLearning.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/MachineLearning.java:37]: I like how the directory creation is done in Java now. Can the `mkdirs` call be done on `myObj`?

File: src/main/java/frc/robot/components/Vision.java
Comment (mmarchetti) [inline src/main/java/frc/robot/components/Vision.java:44]: We should clean these up to be the heading and distance to the hoop rather than x, y, and r to prevent confusing ourselves when debugging.

**Review by mmarchetti — APPROVED** [review]

**Pull request merged**
